src/ringcentral-web-phone.js

```diff
@@ -1155,6 +1167,242 @@
+            return localNetwork[0];
+        }
+        return sysNetwork;
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    //TODO: Check/test networktype values
+
+
+    function getStat(peer, session){
+       
+        var repeatInterval = session.ua.qosCollectInterval;
+        var qosStatsObj = session.qosStatsObj;
+
+        getStats(peer, function (getStatsResult){
+
+            session.qResult = getStatsResult;
+
+            var network = getNetworkType(session.qResult.connectionType);
+
+            if(qosStatsObj.totalIntervalCount>=0){
+                qosStatsObj.netType = addToMap(session.netTypeObj,network, 0);
+            }
+
+            qosStatsObj.localAddr = session.qResult.connectionType.local.ipAddress[0];
+            qosStatsObj.remoteAddr = session.qResult.connectionType.remote.ipAddress[0];
+            session.qResult.results.forEach(function (item) {
+                if (item.type === 'ssrc' && item.transportId === 'Channel-audio-1' && item.id.includes('recv')) {
+                    qosStatsObj.jitterBufferDiscardRate = item.googSecondaryDiscardedRate||0;
+                    qosStatsObj.packetLost = item.packetsLost;
+                    qosStatsObj.packetsReceived = item.packetsReceived;
+                    // qosStatsObj.jitterArr.push(parseFloat(item.googJitterBufferMs));
+                    qosStatsObj.totalSumJitter += parseFloat(item.googJitterBufferMs);
+                    qosStatsObj.totalIntervalCount += 1;
+                    qosStatsObj.JBM = Math.max(qosStatsObj.JBM, parseFloat(item.googJitterBufferMs));
+                }
+            });
+            // qosStatsObj.totalIntervalCount += 1;
+
+        }, repeatInterval);
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    function calculateNetworkUsage(session) {
+            var networkType = [];
+            for (var [key, value] of Object.entries(session.netTypeObj)) {
+                networkType.push(key + ':' + ( value *100 / session.qosStatsObj.totalIntervalCount));
+            }
+            return networkType.join();
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    function calculateStats(session){
+        var qosStatsObj = session.qosStatsObj;
+
+        var rawNLR = qosStatsObj.packetLost* 100 / (qosStatsObj.packetsReceived+qosStatsObj.packetLost);
+        //NLR
+        qosStatsObj.NLR =  parseFloat(rawNLR || 0).toFixed(2);
+        //JitterBufferNominal
+        qosStatsObj.JBN = parseFloat(qosStatsObj.totalSumJitter / qosStatsObj.totalIntervalCount).toFixed(2);
+        //JitterBufferDiscardRate
+        qosStatsObj.JDR =  parseFloat(qosStatsObj.jitterBufferDiscardRate).toFixed(2);
+        //MOS Score
+        qosStatsObj.MOSLQ = 0;
+
```

Comment:
You need to return an object instead of mutating the original:

```js
return Object.assign({}, session.qosStatsObj, {
       NLR:  parseFloat(rawNLR || 0).toFixed(2),
        //JitterBufferNominal
       JBN: parseFloat(qosStatsObj.totalSumJitter / qosStatsObj.totalIntervalCount).toFixed(2),
        //JitterBufferDiscardRate
       JDR:  parseFloat(qosStatsObj.jitterBufferDiscardRate).toFixed(2),
        //MOS Score
       MOSLQ: 0
});
```

Reply:
Done